docker/importer/importer.py

```diff
@@ -455,6 +476,98 @@ def thread_init():
 
     logging.info('Finished processing bucket: %s', source_repo.name)
 
+  def _process_deletions_bucket(self, source_repo: osv.SourceRepository):
+    """Process deletions from a GCS bucket source.
+
+    This validates the continued existence of every Bug in Datastore (for the
+    given source) against every bug currently in that source's GCS bucket,
+    calculating the delta. The bugs determined to have been
+    deleted from GCS are then flagged for deletion from Datastore by the worker.
+
+    If the delta is too large, something undesirable has been assumed to have
+    happened and deletion is aborted.
+    """
+
+    logging.info("Begin processing bucket for deletions: %s", source_repo.name)
+
+    # Get all the existing non-withdrawn Bug IDs for
+    # source_repo.name in Datastore
+    query = osv.Bug.query()
+    query = query.filter(osv.Bug.source == source_repo.name)
+    result = list(query.fetch(keys_only=False))
+    result.sort(key=lambda r: r.id())
+    vuln_ids_for_source = [
+        (r.id(), r.source_id.split(':')[1]) for r in result if not r.withdrawn
```

Comment:
nit: use a namedtuple here for readability.

Reply:
Done in 3bfdddd